vertx-grpc-client/src/main/java/io/vertx/grpc/client/VertxClientCall.java

```diff
@@ -121,6 +123,12 @@ public void start(Listener<ResponseT> responseListener, Metadata headers) {
           }
         });
         writeAdapter.init(request, new BridgeMessageEncoder<>(methodDescriptor.getRequestMarshaller(), compressor));
+      } else {
+        //Maybe the client.request will fail, eg: server interrupt
+        Throwable err = ar1.cause();
+        if (err instanceof ConnectException || err instanceof ClosedChannelException){
```

Comment:
I think we already have error handling in the second asynchronous result that performs a close operation, we should try merging the two in a `compose` operation and handle a single doClose in the composed future.